app/models/kuroko2/job_definition.rb

```diff
@@ -100,6 +101,17 @@ def create_instance(script: nil, launched_by:, token: nil )
     job_instances.create!(script: script, log_message: message)
   end
 
+  def save_and_record_revision(edited_user: nil)
+    record_revision(edited_user: edited_user)
+    save
+  end
+
+  def update_and_record_revision(attributes, edited_user: nil)
```

Comment:
Is this really worked in transaction? I don't think so 🤔 
L105 seems to work well because job_definition is not saved when script revision assignment occurs.

```
irb(main):008:0> jd = JobDefinition.last
  JobDefinition Load (0.4ms)  SELECT  `job_definitions`.* FROM `job_definitions` ORDER BY `job_definitions`.`id` DESC LIMIT 1
=> #<JobDefinition id: 1, script: "a", created_at: "2018-04-27 06:42:42", updated_at: "2018-04-27 06:42:42">
irb(main):009:0> jd.assign_attributes(script: 'updated')
=> nil
irb(main):010:0> jd.script_revisions << ScriptRevision.new(script: 'updated')
   (0.2ms)  BEGIN
  ScriptRevision Create (6.1ms)  INSERT INTO `script_revisions` (`script`, `job_definition_id`, `created_at`, `updated_at`) VALUES ('updated', 1, '2018-04-27 06:47:05', '2018-04-27 06:47:05')
   (0.5ms)  COMMIT
  ScriptRevision Load (26.8ms)  SELECT  `script_revisions`.* FROM `script_revisions` WHERE `script_revisions`.`job_definition_id` = 1 LIMIT 11
=> #<ActiveRecord::Associations::CollectionProxy [#<ScriptRevision id: 1, script: "ab", job_definition_id: 1, created_at: "2018-04-27 06:42:42", updated_at: "2018-04-27 06:42:42">, #<ScriptRevision id: 2, script: "ccccc", job_definition_id: 1, created_at: "2018-04-27 06:43:15", updated_at: "2018-04-27 06:43:15">, #<ScriptRevision id: 3, script: "ccbsadf", job_definition_id: 1, created_at: "2018-04-27 06:43:34", updated_at: "2018-04-27 06:43:34">, #<ScriptRevision id: 4, script: "updated", job_definition_id: 1, created_at: "2018-04-27 06:47:05", updated_at: "2018-04-27 06:47:05">]>
irb(main):011:0> jd.save
   (0.2ms)  BEGIN
  JobDefinition Update (6.8ms)  UPDATE `job_definitions` SET `script` = 'updated', `updated_at` = '2018-04-27 06:47:18' WHERE `job_definitions`.`id` = 1
   (0.8ms)  COMMIT
=> true
```

Reply:
I fixed in dbb501ebf76d44681ea4a8abe68b6ff690f06790.